src/commands/hug.js

```diff
@@ -1,8 +1,66 @@
+import axios from 'axios';
+import Discord from 'discord.js';
 import DiscordCommand from './command.js';
 
 export default class HugCommand extends DiscordCommand {
-    async execute() {
-        const hug = this.getEmojiByName('HugSpin');
-        this.message.channel.send(`${hug}`);
+
+  async execute() {
+    if (!this.member) {
+      this.send("You must be in a Roo Bot enabled Discord server to use this command");
+      return;
+    }
+
+    const hug = this.getEmojiByName('HugSpin');
+    this.message.react(hug);
+
+    let receivers = "";
+    const mentionedUsernames = [];
+
+    if (this.hasRoleMentions || this.hasChannelMentions) {
+      this.send(`Please use \`\`\`${DiscordCommand.prefix}hug @user @user...\`\`\``);
+      return;
     }
+
+    if (this.hasUserMentions) {
+      if (this.args.length == this.userMentions.size) {
+        this.userMentions.each(user => {
```

Comment:
I believe this is legacy API and should follow the more modern Enumerable APIs i.e. `forEach`, `map`, `reduce`, etc.

I would just test this first.

```suggestion
        this.userMentions.forEach(user => {
```